specutils/io/asdf/tags/tests/test_spectra.py

```diff
@@ -1,71 +1,69 @@
+import asdf
+import numpy as np
 import pytest
-# Make sure these tests do not run if ASDF is not installed
-pytest.importorskip('asdf')
+from astropy import units as u
+from astropy.coordinates import FK5
+from astropy.nddata import StdDevUncertainty
 
-import numpy as np  # noqa: E402
-
-import astropy.units as u  # noqa: E402
-from astropy.coordinates import FK5  # noqa: E402
-from astropy.nddata import StdDevUncertainty  # noqa: E402
-
-from asdf.testing.helpers import roundtrip_object  # noqa: E402
-import asdf  # noqa: E402
-
-from specutils import Spectrum1D, SpectrumList, SpectralAxis  # noqa: E402
-from specutils.io.asdf.tags.spectra import Spectrum1DType, SpectrumListType  # noqa: E402
+from specutils import Spectrum1D, SpectrumList, SpectralAxis
+from specutils.io.asdf.tags.tests.helpers import (
+    assert_spectrum1d_equal, assert_spectrumlist_equal)
 
 
 def create_spectrum1d(xmin, xmax, uncertainty=None):
     flux = np.ones(xmax - xmin) * u.Jy
-    wavelength = np.arange(xmin, xmax) * 0.1 * u.nm
+    wavelength = np.arange(xmin, xmax) * u.AA
     uncertainty = StdDevUncertainty(np.ones(xmax - xmin) * u.Jy) if uncertainty is not None else None
-    return Spectrum1D(spectral_axis=wavelength, flux=flux,
-                      uncertainty=uncertainty)
-
+    return Spectrum1D(spectral_axis=wavelength, flux=flux, uncertainty=uncertainty)
 
-@pytest.mark.filterwarnings('ignore:ASDF functionality for astropy is being moved out')
-def test_asdf_spectrum1d(tmpdir):
 
-    spectrum = create_spectrum1d(5100, 5300)
-    Spectrum1DType.assert_equal(spectrum, roundtrip_object(spectrum))
+@pytest.mark.parametrize('uncertainty', [False, True])
+def test_asdf_spectrum1d(tmp_path, uncertainty):
+    file_path = tmp_path / "test.asdf"
+    spectrum = create_spectrum1d(5100, 5300, uncertainty=uncertainty)
+    with asdf.AsdfFile() as af:
+        af["spectrum"] = spectrum
```

Comment:
@WilliamJamieson , I applied your suggestions and schema lookup error is gone. However, now it fails to serialize here. I don't know why the existing serialization code does not work.